plugins/ingest/src/main/java/org/elasticsearch/plugin/ingest/transport/simulate/SimulateProcessorResult.java

```diff
@@ -0,0 +1,128 @@
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.elasticsearch.plugin.ingest.transport.simulate;
+
+import org.elasticsearch.ElasticsearchException;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.common.io.stream.Streamable;
+import org.elasticsearch.common.xcontent.ToXContent;
+import org.elasticsearch.common.xcontent.XContentBuilder;
+import org.elasticsearch.common.xcontent.XContentBuilderString;
+import org.elasticsearch.ingest.Data;
+import org.elasticsearch.plugin.ingest.transport.TransportData;
+
+import java.io.IOException;
+import java.util.Objects;
+
+public class SimulateProcessorResult implements Streamable, ToXContent {
+
+    private String processorId;
+    private TransportData data;
+    private Throwable failure;
+
+    public SimulateProcessorResult() {
+
+    }
+
+    public SimulateProcessorResult(String processorId, Data data) {
+        this.processorId = processorId;
+        this.data = new TransportData(data);
+    }
+
+    public SimulateProcessorResult(String processorId, Throwable failure) {
+        this.processorId = processorId;
+        this.failure = failure;
+    }
+
+    private boolean isFailed() {
+        return this.failure != null;
+    }
+
+    public Data getData() {
+        return data.get();
+    }
+
+    public String getProcessorId() {
+        return processorId;
+    }
+
+    @Override
+    public void readFrom(StreamInput in) throws IOException {
+        boolean isFailure = in.readBoolean();
+        this.processorId = in.readString();
+        if (isFailure) {
+            this.failure = in.readThrowable();
+        } else {
+            this.data = new TransportData();
+            this.data.readFrom(in);
+        }
+    }
+
+    @Override
+    public void writeTo(StreamOutput out) throws IOException {
+        out.writeBoolean(isFailed());
+        out.writeString(processorId);
+        if (isFailed()) {
+            out.writeThrowable(failure);
+        } else {
+            data.writeTo(out);
+        }
+    }
+
+    @Override
+    public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
+        builder.startObject();
+        builder.field(Fields.PROCESSOR_ID, processorId);
+        if (isFailed()) {
+            ElasticsearchException.renderThrowable(builder, params, failure);
+        } else {
+            data.toXContent(builder, params);
+        }
+        builder.endObject();
+        return builder;
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+        if (obj == this) {
+            return true;
+        }
+        if (obj == null || getClass() != obj.getClass()) {
+            return false;
+        }
+        SimulateProcessorResult other = (SimulateProcessorResult) obj;
+        boolean failureEquals = false;
```

Comment:
This needs cleaning up.

testing whether two Throwable objects are equal is not working on the object level, so I decided to just test whether they are of the same class type